source/core/material/warp.cpp

```diff
@@ -59,6 +59,13 @@ namespace pov
 
 const DBL COORDINATE_LIMIT = 1.0e17;
 
+enum BlackHoleType
+{
+    kBlackHoleSpherical_Original=0,
+    kBlackHoleSpherical_Clamped,
+    kBlackHoleSpherical_WhatMightBeUseful
```

Comment:
Thanks for the clarification. I was going to yank all of them. I'd originally planned to do at least one more type, but never got around to it. Commit for the change just pushed.

This req is today against the master branch. A time prior I tried to move another pull request over to a different branch via github and the process failed...  I am going to leave the req aimed as is hoping you know how to make this branch change work. Let me know if the branch change is trouble for you too and I'll create a new pull request against 3.8.0 for this update.